### PR TITLE
Wrap entrypoints into IIFE to protect their variables

### PR DESCRIPTION
**Pull request by weirdan**
Fixes vimeo/psalm#5359

### CODE REVIEW

File: src/Psalm/IssueBuffer.php [hunk 1]
```diff
@@ -542,7 +544,7 @@ function (IssueData $d1, IssueData $d2) : int {
         }
 
         $source_control_info = null;
-        $build_info = (new \Psalm\Internal\ExecutionEnvironment\BuildInfoCollector($_SERVER))->collect();
```
Comment (weirdan) [inline src/Psalm/IssueBuffer.php:545]: This caused issues when `$_SERVER` was destroyed by autoloader

File: tests/EndToEnd/PsalmRunnerTrait.php [hunk 2]
```diff
@@ -42,7 +42,7 @@ private function runPsalm(
             $process->mustRun();
         } else {
             $process->run();
-            $this->assertGreaterThan(0, $process->getExitCode());
+            $this->assertEquals(2, $process->getExitCode());
```
Comment (weirdan) [inline tests/EndToEnd/PsalmRunnerTrait.php:45]: As Psalm now always return exit code 2 when it detects issues, we expect that when `$shouldFail === true`, and let unexpected exceptions be reported as test failures.

File: src/psalm.php [hunk 3]
```diff
@@ -56,810 +57,830 @@
...
     /**
-     * @param string $arg
+     * @param array<int,string> $argv
+     *
+     * Note: the $argv parameter should not be renamed as it's used in the files this function includes via require_once
      */
-    function ($arg) use ($valid_long_options, $valid_short_options): void {
-        if (substr($arg, 0, 2) === '--' && $arg !== '--') {
-            $arg_name = preg_replace('/=.*$/', '', substr($arg, 2));
+    function (array $argv): void {
```
Comment (weirdan) [inline src/psalm.php:72]: We should probably move those closures into methods proper, e.g. `Psalm\Internal\Cli\Psalter::run()`

**Comment by muglug** [comment]
Thanks, this never occurred to me as a potential problem, stupidly

**Pull request merged**